code/modules/food_and_drinks/kitchen_machinery/griddle.dm

```diff
@@ -26,6 +26,20 @@
 	. = ..()
 	grill_loop = new(list(src), FALSE)
 	variant = rand(1,3)
+	RegisterSignal(src, COMSIG_ATOM_EXPOSE_REAGENT, .proc/on_expose_reagent)
+
+/obj/machinery/griddle/proc/on_expose_reagent(atom/parent_atom, datum/reagent/exposing_reagent, reac_volume)
+	SIGNAL_HANDLER
+
+	if(griddled_objects.len >= max_items || !istype(exposing_reagent, /datum/reagent/consumable/pancakebatter) || reac_volume < 5)
```

Comment:
:-1:
Make this a proc on the reagent itself, this behavior shouldn't be on the griddle.

Reply:
adding a proc on reagents for each object a reagent might have interactions with is anything but modular. If we started making it possible to have like 10 reagents you can make items with on the griddle we can consider it, but there is no need for this to be on the reagent right now.

Reply:
I talked with floyd about it, we agreed this wasn't big enough to add onto reagents